modules/sdk-coin-coreum/test/unit/coreum.ts

```diff
@@ -388,4 +392,129 @@ describe('Coreum', function () {
       stub.restore();
     });
   });
+
+  describe('Recover transaction: success path', () => {
+    const sandBox = sinon.createSandbox();
+    const destinationAddress = wrwUser.destinationAddress;
+    const coin = coins.get('tcoreum');
+    const testBalance = '1500000';
+    const testAccountNumber = '123';
+    const testSequenceNumber = '0';
+    const testChainId = 'test-chain';
+
+    beforeEach(() => {
+      const accountBalance = sandBox.stub(Tcoreum.prototype, 'getAccountBalance' as keyof Tcoreum);
+      accountBalance.withArgs(wrwUser.senderAddress).resolves(testBalance);
+
+      const accountDetails = sandBox.stub(Tcoreum.prototype, 'getAccountDetails' as keyof Tcoreum);
+      accountDetails.withArgs(wrwUser.senderAddress).resolves([testAccountNumber, testSequenceNumber]);
+
+      const chainId = sandBox.stub(Tcoreum.prototype, 'getChainId' as keyof Tcoreum);
+      chainId.withArgs().resolves(testChainId);
+
+      const deserializedEntChallenge = EcdsaTypes.deserializeNtildeWithProofs(mockSerializedChallengeWithProofs);
+      sinon.stub(EcdsaRangeProof, 'generateNtilde').resolves(deserializedEntChallenge);
+    });
+
+    afterEach(() => {
+      sandBox.restore();
+      sinon.restore();
+    });
+
+    it('should recover funds for non-bitgo recoveries', async function () {
+      const res = await tcoreum.recover({
+        userKey: wrwUser.userPrivateKey,
+        backupKey: wrwUser.backupPrivateKey,
+        bitgoKey: wrwUser.bitgoPublicKey,
+        walletPassphrase: wrwUser.walletPassphrase,
+        recoveryDestination: destinationAddress,
+      });
+      res.should.not.be.empty();
+      res.should.hasOwnProperty('serializedTx');
+      sandBox.assert.calledOnce(tcoreum.getAccountBalance);
+      sandBox.assert.calledOnce(tcoreum.getAccountDetails);
+      sandBox.assert.calledOnce(tcoreum.getChainId);
+
+      const tcoreumTxn = new CosmosTransaction(coin, testnetUtils);
+      tcoreumTxn.enrichTransactionDetailsFromRawTransaction(res.serializedTx);
+      const tcoreumTxnJson = tcoreumTxn.toJson();
+      const sendMessage = tcoreumTxnJson.sendMessages[0].value as SendMessage;
+      const balance = new BigNumber(testBalance);
+      const actualBalance = balance.minus(new BigNumber(6250));
```

Comment:
seems to be a magic number; where does it come from and can it be a constant ?

Reply:
updated